### PR TITLE
feat: implement global state for fetcher

### PR DESCRIPTION
**Pull request by mnindrazaka**

### CODE REVIEW

File: src/utils/fetcher/QueryReducer.ts [hunk 1]
```diff
@@ -1,52 +1,57 @@
...
-  | { tag: "idle" }
-  | { tag: "loading" }
-  | { tag: "success"; data: T }
-  | { tag: "error"; error: Error };
+export namespace State {
+  export type t<T> =
+    | { tag: "idle" }
+    | { tag: "loading" }
+    | { tag: "success"; data: T }
+    | { tag: "error"; error: Error };
```
Comment (AndriP8) [inline src/utils/fetcher/QueryReducer.ts:9]: Type `error: Error` ini darimana yaa?

Comment (mnindrazaka) [reply src/utils/fetcher/QueryReducer.ts:9]: type Error itu bawaan typescript

Comment (mnindrazaka) [reply src/utils/fetcher/QueryReducer.ts:9]: sorry, javascript ternyata :   
https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error

File: src/utils/fetcher/QueryContext.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import * as QueryReducer from "./QueryReducer";
+
+export namespace Store {
+  export type t = Record<string, QueryReducer.State.t<any>>;
```
Comment (AndriP8) [inline src/utils/fetcher/QueryContext.tsx:5]: Ini cara bacanya gimana yaa? `Record<string, QueryReducer.State.t<any>>`

Comment (mnindrazaka) [reply src/utils/fetcher/QueryContext.tsx:5]: `Record` itu sama kayak gini : 

```typescript
type t =  { [string]: QueryReducer.State.t<any> }
```

Comment (AndriP8) [reply src/utils/fetcher/QueryContext.tsx:5]: Okee, paham paham

File: src/utils/fetcher/QueryReducer.ts (hunk 1 above)
Comment (AndriP8) [reply src/utils/fetcher/QueryReducer.ts:9]: Oooo, baru tau ternyata bawaan javascript

**Review by AndriP8 — APPROVED** [review]

**Pull request merged**